### PR TITLE
Fix cmake crowd mfg 2d and small nits

### PR DESCRIPTION
**Pull request by TheoCabannes**
I found that mfg_crowd_modelling_2d is not in the CMake file. Therefore open_spiel/python/mfg/games/crowd_modelling_example.py cannot be run.
This PR adds crowd mfg 2d to the CMake and fix two nits: some `;` forgotten in the tests and no definition of `ProcessStringParam` in the header file (it is used in the test file).

Currently the setup.py install works. But the build in `build_and_run_tests.sh` fails and I do not know why. Hopefully somebody can help me.

### CODE REVIEW

File: open_spiel/games/mfg/crowd_modelling_2d.h [hunk 1]
```diff
@@ -51,6 +51,26 @@ inline constexpr const char* kInitialDistributionValue = "[]";  // "[0.5;0.5]"
 // Action that leads to no displacement on the torus of the game.
 inline constexpr int kNeutralAction = 2;
 
+std::vector<absl::string_view> ProcessStringParam(
+    const std::string& string_param_str, int max_size) {
+  // ProcessStringParam takes a parameter string and split it is a sequence of
```
Comment (RaphaelMarinier) [inline open_spiel/games/mfg/crowd_modelling_2d.h:56]: You should keep the implementation of this function in the .cc file.

File: open_spiel/games/mfg/crowd_modelling_2d_test.cc [hunk 2]
```diff
@@ -112,13 +112,13 @@ void TestReward() {
...
-  auto split_string_list1 = ProcessStringParam("[0|0;0|1]", 5)
+  auto split_string_list1 = ProcessStringParam("[0|0;0|1]", 5);
   SPIEL_CHECK_EQ(split_string_list1.size(), 2);
-  auto split_string_list2 = ProcessStringParam("[0|2;0|3;0|4]", 5)
+  auto split_string_list2 = ProcessStringParam("[0|2;0|3;0|4]", 5);
   SPIEL_CHECK_EQ(split_string_list2.size(), 3);
-  auto split_string_list3 = ProcessStringParam("[0.5;0.5]", 5)
+  auto split_string_list3 = ProcessStringParam("[0.5;0.5]", 5);
   SPIEL_CHECK_EQ(split_string_list3.size(), 2);
 }
```
Comment (RaphaelMarinier) [inline open_spiel/games/mfg/crowd_modelling_2d_test.cc:124]: Hum, it seems this test was never run. You need to use the namespace crowd_modelling_2d in the main() below.

**Review by RaphaelMarinier — CHANGES_REQUESTED** [review]

**Comment by TheoCabannes** [comment]
Thank you a lot @RaphaelMarinier, I took your comments into account. The build now works, I have not caught your second comments when working on the code.

The test `games_sim_test.py` is failing due to
```
======================================================================
ERROR: test_game_sim_mfg_crowd_modelling_2d (__main__.GamesSimTest)
GamesSimTest.test_game_sim_mfg_crowd_modelling_2d
test_game_sim_mfg_crowd_modelling_2d(<GameType 'mfg_crowd_modelling_2d'>)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/theophile/miniconda3/envs/dev/lib/python3.7/site-packages/absl/testing/parameterized.py", line 316, in bound_param_test
    return test_method(self, *testcase_params)
  File "/Users/theophile/Documents/research/Markov model/open_spiel/open_spiel/python/tests/games_sim_test.py", line 201, in test_game_sim
    self.sim_game(game)
  File "/Users/theophile/Documents/research/Markov model/open_spiel/open_spiel/python/tests/games_sim_test.py", line 136, in sim_game
    check_pickle_serialization)
  File "/Users/theophile/Documents/research/Markov model/open_spiel/open_spiel/python/tests/games_sim_test.py", line 93, in serialize_deserialize
    ser_str = pyspiel.serialize_game_and_state(game, state)
pyspiel.SpielError: /Users/theophile/Documents/research/Markov model/open_spiel/open_spiel/spiel.cc:307 game_->GetType().dynamics != GameType::Dynamics::kMeanField
game_->GetType().dynamics = MeanField, GameType::Dynamics::kMeanField = MeanField
```

**Comment by RaphaelMarinier** [comment]
The 2d game is missing serialization logic (see comment [there](https://github.com/deepmind/open_spiel/blob/cfe17d1e013a1f74aa9433139a58eba8d8afd3b7/open_spiel/spiel.cc#L299), and compare with the 1D crowd modelling which [has the serialization logic](https://github.com/deepmind/open_spiel/blob/cfe17d1e013a1f74aa9433139a58eba8d8afd3b7/open_spiel/games/mfg/crowd_modelling.cc#L224)).

@perolat is working on it.

**Comment by TheoCabannes** [comment]
> The 2d game is missing serialization logic, which explains this failure (see comment [there](https://github.com/deepmind/open_spiel/blob/cfe17d1e013a1f74aa9433139a58eba8d8afd3b7/open_spiel/spiel.cc#L299), and compare with the 1D crowd modelling which [has the serialization logic](https://github.com/deepmind/open_spiel/blob/cfe17d1e013a1f74aa9433139a58eba8d8afd3b7/open_spiel/games/mfg/crowd_modelling.cc#L224)).
> 
> @perolat is working on it.

Should I remove the PR, or should I remove the tests, or should we keep the PR with the failing tests?
It seems that removing the tests make the more sense so the PR can be checked in.

**Comment by RaphaelMarinier** [comment]
Indeed, I think the best is that you exclude the game from the tests for now ([here](https://github.com/deepmind/open_spiel/blob/cfe17d1e013a1f74aa9433139a58eba8d8afd3b7/open_spiel/python/tests/games_sim_test.py#L38), and in other places if needed), so that you can submit your PR, without making tests fails.

**Comment by TheoCabannes** [comment]
I have disabled the game sim test. I have disabled one failing test in crowd_modelling_2d_test. I have assigned re-enabling the tests to @perolat as you told me that he is working on it.

**Comment by lanctot** [comment]
Hi guys, I have imported this, but I removed the custom disabling of tests, because it turned out there were even more places we had to add stuff and it was not clean. Instead, I just didn't register mfg_crowd_modelling_2d. 

I think that's fine to do the custom disabling of tests while you're still working on it, but we should leave those on the user side and not check them into the repos. There are also better ways to bypass the tests if the game does not (yet) support serialization. I'll have a chat with Julien and Raphael to explain how to go about this so we can partially include it in the tests in a cleaner way.

**Comment by lanctot** [comment]
Ok we added serialization to mfg 2d and enabled it in the tests fully, in this commit: https://github.com/deepmind/open_spiel/commit/bc55df5173bd49ddf4499df1d8a60e7df8e8b576

**Pull request merged**